src/unix/bsd/netbsdlike/openbsd/x86_64.rs

```diff
@@ -3,6 +3,96 @@ use PT_FIRSTMACH;
 pub type c_long = i64;
 pub type c_ulong = u64;
 pub type c_char = i8;
+pub type ucontext_t = sigcontext;
+
+s_no_extra_traits! {
+    #[repr(C, packed)]
+    pub struct fxsave64 {
+        pub fx_fcw: u16,
+        pub fx_fsw: u16,
+        pub fx_ftw: u8,
+        __unused: u8,
+        pub fx_fop: u16,
+        pub fx_rip: u64,
+        pub fx_rdp: u64,
+        pub fx_mxcsr: u32,
+        pub fx_mxcsr_mask: u32,
+        pub fx_st: [[u64; 2]; 8],
+        pub fx_xmm: [[u64; 2]; 16],
+        __reserved: [u8; 96],
+    }
+}
+
+cfg_if! {
+    if #[cfg(feature = "extra_traits")] {
+        impl PartialEq for fxsave64 {
+            fn eq(&self, other: &fxsave64) -> bool {
+                self.fx_fcw == other.fx_fcw &&
+                self.fx_fsw == other.fx_fsw &&
+                self.fx_ftw == other.fx_ftw &&
+                self.fx_fop == other.fx_fop &&
+                self.fx_rip == other.fx_rip &&
+                self.fx_rdp == other.fx_rdp &&
+                self.fx_mxcsr == other.fx_mxcsr &&
+                self.fx_mxcsr_mask == other.fx_mxcsr_mask &&
+                self.fx_st == other.fx_st &&
+                std::ptr::addr_of!(self.fx_xmm) == std::ptr::addr_of!(other.fx_xmm)
```

Comment:
Doesn't this break the build if a user uses Rust > 1.51.0?

Reply:
it also does not work on all configurations but I still do not know how to make it work for all cases with this packed struct. I really wish it is not a blocker for libc next release tough ? otherwise this PR is not really urgent.

Reply:
We can add a cfg (`libc_addr_of` or something else) like:

https://github.com/rust-lang/libc/blob/74779489b93d060468767a71e1fc2a0de1d63e4c/build.rs#L42-L45

Reply:
>  really wish it is not a blocker for libc next release tough ? otherwise this PR is not really urgent.

Then I'm going to make a new release, thanks :)